x-pack/legacy/plugins/triggers_actions_ui/np_ready/public/plugin.ts

```diff
@@ -0,0 +1,146 @@
+  PluginInitializerContext,
+} from 'src/core/public';
+
+import template from '../../public/index.html';
+import { renderReact } from './application';
+import { BASE_PATH } from './application/constants';
+import { breadcrumbService } from './application/lib/breadcrumb';
+import { docTitleService } from './application/lib/doc_title';
+import { ActionTypeRegistry } from './application/action_type_registry';
+import { registerBuiltInActionTypes } from './application/components/builtin_action_types';
+import { AlertTypeRegistry } from './application/alert_type_registry';
+import { registerBuiltInAlertTypes } from './application/components/builtin_alert_types';
+import { setSavedObjectsClient } from './application/components/builtin_alert_types/threshold/lib/api';
+import { hasShowActionsCapability, hasShowAlertsCapability } from './application/lib/capabilities';
+
+export type Setup = void;
+export type Start = void;
+
+const REACT_ROOT_ID = 'triggersActionsRoot';
+
+export class Plugin implements CorePlugin<Setup, Start> {
+  private actionTypeRegistry: ActionTypeRegistry;
+  private alertTypeRegistry: AlertTypeRegistry;
+
+  constructor(initializerContext: PluginInitializerContext) {
+    const actionTypeRegistry = new ActionTypeRegistry();
+    this.actionTypeRegistry = actionTypeRegistry;
+
+    const alertTypeRegistry = new AlertTypeRegistry();
+    this.alertTypeRegistry = alertTypeRegistry;
+  }
+
+  public setup(core: CoreSetup, plugins: any): Setup {
+    /*
+      The code below would be replacing for current:
+      uiExports: {
+        managementSections: ['myplugin/management'],
+      }
+    */
+    const {
+      capabilities,
+      management: { getSection },
+    } = plugins;
+
+    const canShowActions = hasShowActionsCapability(capabilities.get());
+    const canShowAlerts = hasShowAlertsCapability(capabilities.get());
+    if (canShowActions || canShowAlerts) {
+      registerBuiltInActionTypes({
+        actionTypeRegistry: this.actionTypeRegistry,
+      });
+
+      registerBuiltInAlertTypes({
+        alertTypeRegistry: this.alertTypeRegistry,
+      });
+
+      const kbnSection = getSection('kibana');
+      kbnSection.register('triggersActions', {
+        display: i18n.translate('xpack.triggersActionsUI.managementSection.displayName', {
+          defaultMessage: 'Alerts and actions',
+        }),
+        order: 7,
+        url: `#${BASE_PATH}`,
+      });
+    }
+  }
+
+  public start(core: CoreStart, plugins: any) {
+    const { capabilities } = plugins;
+    const canShowActions = hasShowActionsCapability(capabilities.get());
+    const canShowAlerts = hasShowAlertsCapability(capabilities.get());
+    // AppCore/AppPlugins to be passed on as React context
+    const AppDependencies = {
+      core,
+      plugins,
+      actionTypeRegistry: this.actionTypeRegistry,
+      alertTypeRegistry: this.alertTypeRegistry,
+    };
+
+    // Don't register routes when user doesn't have access to the application
```

Comment:
Is the link in the sidebar hidden as well?

Reply:
Good catch! Fixed here: https://github.com/elastic/kibana/pull/48959/commits/986b6bee84dac5d1cb6d45afa92791f3fef9d3bb.